os/x11/window.cpp

```diff
@@ -1042,18 +1042,18 @@ void WindowX11::processX11Event(XEvent& event)
         KEY_TRACE("Xutf8LookupString %s\n", &buf[0]);
       }
 
+
       // Check if the key has been pressed, 
-      // and if yes - check if it's the same one key
-      // as in the previous event.  If it's the same one
-      // - it means key is being held, so set repeat to 1.
+      // and if yes - check what's the previous state
+      // of the key symbol - if it was previously pressed
+      // set repeat count of 1, if it wasn't pressed previously
+      // it will set repeat count to 0
       if (event.type == KeyPress) {
-        if (keysym == m_pressedKeySym)
+        if (map_keys[keysym])
           ev.setRepeat(1);
-        m_pressedKeySym = keysym;
       }
-      else
-        m_pressedKeySym = 0;
 
+      map_keys[keysym] = event.type == KeyPress ? true : false;
```

Comment:
I'm not sure if the map should be set to false when the key is released or just remove the element from the map.

I think we can have something like:
```c++
      if (event.type == KeyPress) {
        if (g_pressedKeys.find(keysym) != g_pressedKeys.end())
          ev.setRepeat(1);
        else
          g_pressedKeys.insert(keysym);
      }
      else
        g_pressedKeys.erase(keysym);
```

Reply:
Done, also please check out https://github.com/aseprite/aseprite/pull/4080 - this fixes a behaviour when we for example press Shift + Key that got mapped to mouse trigger button, and then we release the key that got mapped but never release Shift. I've explained it in the PR's comments.